swc-uxp-react-starter/README.md

```diff
@@ -0,0 +1,214 @@
+
+```
+import { aliases } from '@swc-uxp-wrappers/utils';
+```
+
+5. Run `yarn build` to prepare the distribution bundle.
+You can also use `yarn watch` to create the bundle as soon as you save your changes to any source files. Use this along with Plugin -> Watch option in UDT to sync with latest changes automatically.
+
+## Deep dive
+Now that the plugin is working, let's look into the details.
+### webpack.config.js
+
+Webpack is used to bundle the dependencies in the project therefore you would see the webpack.config.js file for basic config.
+
+Notice the installed `@swc-uxp-wrappers/utils` package in the package.json file. It is being used it to provide [aliasing](https://webpack.js.org/configuration/resolve/#resolvealias) via mapping. You can remove this aliasing and use the same plugin on web too.
+
+```
+       resolve: {
+            extensions: ['.js', '.json'],
+            alias: aliases,
+        }
+```
+
+Pro tip: For debugging purposes, add `eval-cheap-source-map` in the webpack.config file to get the source map in UDT debug window.
+
+```javascript
+devtool: 'eval-cheap-source-map'
+```
+
+
+### .babelrc
+
+This is config file for `babel` library to transpile the code, especially the JSX syntax, to match the environment's capabilities.
+
+
+### package.json
+
+Once you install the component (using `yarn add`) you should see the components added to the 'dependencies'.
+
+
+### manifest.json
+Enable SWC by setting the **enableSWCSupport** flag to true.
+
+```
+"featureFlags": {
+   "enableSWCSupport": true
+}
+```
+
+### src/index.js
+
+This is the entry point of the project. 
+
+Ensure that the the components are wrapped with a `Theme` component. It ensures that the Spectrum design tokens are delivered to the scoped HTML context.
+```
+const root = ReactDOM.createRoot(document.getElementById("root"));
+root.render(
+  <React.StrictMode>
+    <Theme theme="spectrum" scale="medium" color="light">
+      <App />
+    </Theme>
+  </React.StrictMode>
+);
+
+```
+
+### **src/index.html**
+
+Entry point of the HTML structure of React app index.js.
+
+### **src/App.js**
+The main React component. This component serves as the root of your component tree. You may also wrap your component with `Theme` here.
+```
+function App() {
+  return (
+    <div>
+      <h1>Welcome</h1>
+    <Theme theme="spectrum" scale="medium" color="light">
+        <Menu selects="single">
+          <MenuItem selected>Sample Menu</MenuItem>
+          <MenuDivider></MenuDivider>
+          <MenuItem>Select Option 1</MenuItem>
+          <MenuItem>Select Option 2</MenuItem>
+          <MenuItem>Select Option 3</MenuItem>
+          <MenuItem>Select Option 4</MenuItem>
+          <MenuItem disabled>Disabled Option</MenuItem>
+        </Menu>
+        </Theme>
+        <br />
+        This is an <Link href="#">example link</Link>.
+    </div>
+  );
+}
+
+```
+
+
+## Recommended `@swc-react` library  versions
```

Comment:
Can we publish this list in our public docs and provide the link here? Will avoid duplicates and maintenance.

Reply:
Added the link here